api/test/e2e/ssl_test.go

```diff
@@ -0,0 +1,122 @@
+ * limitations under the License.
+ */
+package e2e
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSSL_Basic(t *testing.T) {
+	testCert, err := ioutil.ReadFile("../certs/test2.crt")
+	assert.Nil(t, err)
+	testKey, err := ioutil.ReadFile("../certs/test2.key")
+	assert.Nil(t, err)
+	apisixKey, err := ioutil.ReadFile("../certs/apisix.key")
+	assert.Nil(t, err)
+	body, err := json.Marshal(map[string]string{
+		"id":   "1",
+		"cert": string(testCert),
+		"key":  string(testKey),
+	})
+	assert.Nil(t, err)
+	invalidBody, err := json.Marshal(map[string]string{
+		"id":   "1",
+		"cert": string(testCert),
+		"key":  string(apisixKey),
+	})
+
+	//Before configuring SSL, make a HTTPS request
+	http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
+	_, err = http.Get("https://www.test2.com:9443")
+	assert.NotNil(t, err)
+	assert.EqualError(t, err, "Get \"https://www.test2.com:9443\": remote error: tls: internal error")
+
+	tests := []HttpTestCase{
+		{
+			caseDesc:     "create ssl fail - key and cert not match",
+			Object:       MangerApiExpect(t),
+			Method:       http.MethodPost,
+			Path:         "/apisix/admin/ssl",
+			Body:         string(invalidBody),
+			Headers:      map[string]string{"Authorization": token},
+			ExpectStatus: http.StatusBadRequest,
+		},
+		{
+			caseDesc:     "create ssl successfully",
+			Object:       MangerApiExpect(t),
+			Method:       http.MethodPost,
+			Path:         "/apisix/admin/ssl",
+			Body:         string(body),
+			Headers:      map[string]string{"Authorization": token},
+			ExpectStatus: http.StatusOK,
+		},
+		{
+			caseDesc: "create route",
+			Object:   MangerApiExpect(t),
+			Method:   http.MethodPut,
+			Path:     "/apisix/admin/routes/r1",
+			Body: `{
+				"uri": "/hello_",
+				"hosts": ["test2.com", "*.test2.com"],
+				"upstream": {
+					"nodes": {
+						"172.16.238.20:1980": 1
+					},
+					"type": "roundrobin"
+				}
+			}`,
+			Headers:      map[string]string{"Authorization": token},
+			ExpectStatus: http.StatusOK,
+		},
+		{
+			caseDesc:     "hit the route just created using HTTPS",
+			Object:       APISIXHTTPSExpect(t),
+			Method:       http.MethodGet,
+			Path:         "/hello_",
+			Headers:      map[string]string{"Host": "www.test2.com"},
+			ExpectStatus: http.StatusOK,
+			ExpectBody:   "hello world\n",
+			Sleep:        sleepTime,
+		},
+		{
+			caseDesc:     "delete route",
+			Object:       MangerApiExpect(t),
+			Method:       http.MethodDelete,
+			Path:         "/apisix/admin/routes/r1",
+			Headers:      map[string]string{"Authorization": token},
+			ExpectStatus: http.StatusOK,
+		},
+		{
+			caseDesc:     "delete ssl",
+			Object:       MangerApiExpect(t),
+			Method:       http.MethodDelete,
+			Path:         "/apisix/admin/ssl/1",
+			Headers:      map[string]string{"Authorization": token},
+			ExpectStatus: http.StatusOK,
```

Comment:
We need to check again after delete route and ssl cert

Reply:
ok